deadlines-api/src/main/java/com/palantir/deadlines/api/DeadlineExpiredException.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.deadlines.api;
+
+public class DeadlineExpiredException extends RuntimeException {}
```

Comment:
thoughts on this? do we want to encapsulate any more data?

Reply:
I think we need a way to distinguish if a deadline was exceeded due to a local constraint, or a shorter provided deadline

Reply:
Might as well implement SafeLoggable with a constant message. Unclear if this will need ctor overloads taking causal throwables

Reply:
i've removed this exception class for now - we can merge a separate PR later to add error types when we want to enforce failures.